server/cache/refDataCache.ts

```diff
@@ -0,0 +1,43 @@
+import { createRedisClient } from '../data/redisClient'
+import logger from '../../logger'
+import config from '../config'
+
+const redisEnabled = config.redis.enabled
+const redis = redisEnabled ? createRedisClient() : null
+if (redisEnabled && redis) {
+  redis.connect().catch(err => logger.error('Redis connect (refDataCache) failed', err))
+} else {
+  logger.info('Redis caching disabled — using API only')
+}
+
+const getFullCacheName = (key: string) => `ref-data:${key}`
+
+export default async function getOrSetJson<T>(
+  cacheKey: string,
+  apiCall: () => Promise<T>,
+  timeToLiveSeconds = 24 * 60 * 60, // 24 hours TTL
+): Promise<T> {
+  if (!config.redis.enabled || !redis) {
+    return apiCall()
+  }
+  logger.info(`Fetching ref data from cache ${cacheKey}`)
```

Comment:
added extra log lines for testing purposes - will remove these in a subsequent PR